bucketeer/src/main/kotlin/io/bucketeer/sdk/android/internal/evaluation/EvaluationInteractor.kt

```diff
@@ -80,49 +80,48 @@ internal class EvaluationInteractor(
+          // 1- Delete all the evaluations from DB, and save the latest evaluations from the response into the DB
+          // 2- Save the UserEvaluations.CreatedAt in the response as evaluatedAt in the SharedPreferences
+          evaluationStorage.deleteAllAndInsert(
+            evaluationsId = newEvaluationsId,
+            evaluations = currentEvaluations,
+            evaluatedAt = newEvaluatedAt,
+          )
+        } else {
+          // 1- Check the evaluation list in the response and upsert them in the DB if the list is not empty
+          // 2- Check the list of the feature flags that were archived on the console and delete them from the DB
+          // 3- Save the UserEvaluations.CreatedAt in the response as evaluatedAt in the SharedPreferences
+          val archivedFeatureIds = response.evaluations.archivedFeatureIds
+          val updatedEvaluations = response.evaluations.evaluations
+          shouldNotifyListener =
+            evaluationStorage.update(
               evaluationsId = newEvaluationsId,
-              evaluations = currentEvaluations,
+              evaluations = updatedEvaluations,
+              archivedFeatureIds = archivedFeatureIds,
               evaluatedAt = newEvaluatedAt,
             )
-          } else {
-            // 1- Check the evaluation list in the response and upsert them in the DB if the list is not empty
-            // 2- Check the list of the feature flags that were archived on the console and delete them from the DB
-            // 3- Save the UserEvaluations.CreatedAt in the response as evaluatedAt in the SharedPreferences
-            val archivedFeatureIds = response.evaluations.archivedFeatureIds
-            val updatedEvaluations = response.evaluations.evaluations
-            shouldNotifyListener =
-              evaluationStorage.update(
-                evaluationsId = newEvaluationsId,
-                evaluations = updatedEvaluations,
-                archivedFeatureIds = archivedFeatureIds,
-                evaluatedAt = newEvaluatedAt,
-              )
-          }
-        } catch (ex: Exception) {
-          loge { "Failed to update latest evaluations" }
-          return GetEvaluationsResult.Failure(
-            BKTException.IllegalStateException("error: ${ex.message}"),
-            featureTag,
-          )
         }
 
         evaluationStorage.clearUserAttributesUpdated()
         // Update listeners should be called on the main thread
         // to avoid unintentional lock on Interactor's execution thread.
         if (shouldNotifyListener) {
           mainHandler.post {
-            updateListeners.forEach { it.value.onUpdate() }
+            updateListeners.forEach {
+              // Prevent crash if consumer code throwing unhandled error
+              runCatching {
+                it.value.onUpdate()
+              }.onFailure { onUpdateError ->
+                logd(onUpdateError) { "onUpdateError: ${onUpdateError.message}" }
```

Comment:
```suggestion
                loge(onUpdateError) { "failed while calling onUpdate listener: ${onUpdateError.message}" }
```

Reply:
Can you implement a way to report this as an internal error, too?
Since this interface is only called when the cache is updated, we could report the API ID as `GET_EVALUATIONS`.
WDYT?

Reply:
Yes, we can.
However, since this code runs on the main thread, reporting this error is not straightforward